src/server/run_sched.c

```diff
@@ -155,6 +159,132 @@ put_sched_cmd(int sock, int cmd, char *jobid)
 	return -1;
 }
 
+/**
+ * @brief
+ * 		find_assoc_sched - find the corresponding scheduler which is responsible
+ * 		for handling this job.
+ *
+ * @param[in]	jid - job id
+ * @param[out]	target_sched - pointer to the corresponding scheduler to which the job belongs to
+ *
+ * @retval - 1  if success
+ * 	   - 0 if fail
+ */
+int
+find_assoc_sched_jid(char *jid, pbs_sched **target_sched)
```

Comment:
why don't we expect a job structure to be passed to the function. It will probably save you a find job call.
I am getting a very bleak memory that I have asked this before too :)

Reply:
Not all the places where we call this function job structure is not available in which case we any way have to do find_job() call